spyder/plugins/outlineexplorer/widgets.py

```diff
@@ -608,8 +608,7 @@ def update_tree(self, items, editor):
         if not must_update:
             # Compare with current tree to check if it's necessary to update
             # it.
-            changes = tree - current_tree
-            if tree and len(changes) == 0:
+            if tree and tree == current_tree:
```

Comment:
I am not sure if we still need to check that the tree is not empty?

Reply:
I think is not needed anymore :+1:

Reply:
Done in https://github.com/spyder-ide/spyder/pull/21481/commits/c3606519d3f5015a4be5123ee72c624c91b44e53.